Lib/test/test_email/test_email.py

```diff
@@ -3319,32 +3319,92 @@ def test_getaddresses(self):
            [('Al Person', 'aperson@dom.ain'),
             ('Bud Person', 'bperson@dom.ain')])
 
-    def test_getaddresses_comma_in_name(self):
-        """GH-106669 regression test."""
-        self.assertEqual(
-            utils.getaddresses(
-                [
-                    '"Bud, Person" <bperson@dom.ain>',
-                    'aperson@dom.ain (Al Person)',
-                    '"Mariusz Felisiak" <to@example.com>',
-                ]
-            ),
-            [
-                ('Bud, Person', 'bperson@dom.ain'),
-                ('Al Person', 'aperson@dom.ain'),
-                ('Mariusz Felisiak', 'to@example.com'),
-            ],
-        )
+    def test_getaddresses_parsing_errors(self):
+        """Test for parsing errors from CVE-2023-27043"""
+        eq = self.assertEqual
+        eq(utils.getaddresses(['alice@example.org(<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org)<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org<<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org><bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org@<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org,<bob@example.com>']),
+           [('', 'alice@example.org'), ('', 'bob@example.com')])
+        eq(utils.getaddresses(['alice@example.org;<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org:<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org.<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org"<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org[<bob@example.com>']),
+           [('', '')])
+        eq(utils.getaddresses(['alice@example.org]<bob@example.com>']),
+           [('', '')])
```

Comment:
```suggestion
        eq(utils.getaddresses(['alice@example.org]<bob@example.com>']),
           [('', '')])
        eq(utils.getaddresses(['alice@example.org])(<bob@example.com>']),
           [('', '')])
```

Reply:
However, not a bad idea to add extra test. That way if someone else changes this code in the future these other corner cases will be tested for.